job.go

```diff
@@ -45,46 +53,55 @@ func (j *Job) shouldRun() bool {
 }
 
 //Run the job and immediately reschedule it
-func (j *Job) run() (result []reflect.Value, err error) {
+func (j *Job) run() ([]reflect.Value, error) {
 	if j.lock {
 		if locker == nil {
-			err = fmt.Errorf("trying to lock %s with nil locker", j.jobFunc)
-			return
+			return nil, fmt.Errorf("trying to lock %s with nil locker", j.jobFunc)
 		}
 		key := getFunctionKey(j.jobFunc)
 
-		if ok, err := locker.Lock(key); err != nil || !ok {
-			return nil, err
-		}
-
-		defer func() {
-			if e := locker.Unlock(key); e != nil {
-				err = e
-			}
-		}()
+		locker.Lock(key)
+		defer locker.Unlock(key)
 	}
 
 	j.lastRun = time.Now()
-	result, err = callJobFuncWithParams(j.funcs[j.jobFunc], j.fparams[j.jobFunc])
-	j.scheduleNextRun()
-	return
+	if err := j.scheduleNextRun(); err != nil {
```

Comment:
attention here: I swiped execution logic -> scheduleNextRun happens before calljobWithParams now. 

In the earlier implementation, if callJobWithParams takes long to execute, next run schedule will be impacted. I thought we had fixed this in the past (?)